common/src/main/java/com/wynntils/services/mapdata/providers/builtin/LootChestsProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.mapdata.providers.builtin;
+
+import com.wynntils.models.containers.type.LootChestTier;
+import com.wynntils.services.mapdata.providers.json.JsonMapLocation;
+import com.wynntils.services.mapdata.type.MapFeature;
+import com.wynntils.utils.mc.type.Location;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class LootChestsProvider extends BuiltInProvider {
+    private static final List<MapFeature> PROVIDED_FEATURES = new ArrayList<>();
+
+    @Override
+    public String getProviderId() {
+        return "loot-chests";
+    }
+
+    @Override
+    public Stream<MapFeature> getFeatures() {
+        return PROVIDED_FEATURES.stream();
+    }
+
+    public void updateFoundChests(List<FoundChestLocation> foundChests) {
+        PROVIDED_FEATURES.forEach(this::notifyCallbacks);
+        PROVIDED_FEATURES.clear();
+        foundChests.forEach(LootChestsProvider::registerFeature);
+    }
+
+    public static void registerFeature(FoundChestLocation foundChest) {
+        PROVIDED_FEATURES.add(foundChest);
+    }
+
+    public static final class FoundChestLocation extends JsonMapLocation {
```

Comment:
It seems a bit odd to have this extend "Json" MapLocation. I had to check the code to understand why. I see now that there are several "Json" implementations that are basic implementations of interfaces which just maps getters to fields, and sets those using a constructor. That kind of implementation is good, but maybe we should not call it "Json". 

I'm not sure what to rename it to, though. Something like "BasicMapLocation" or "SimpleMapLocation", maybe? "ConcreteMapLocation"?

Reply:
How would the new class look like? I thought it's obvious that we have to use a json map location here, so these can be serialized/deserialized by GSON.

Reply:
Ok, I see how you mean. I didn't really think about it like that, but sure, I can agree with that.